services/src/main/java/org/keycloak/authentication/DefaultAuthenticationFlow.java

```diff
@@ -48,7 +49,7 @@
     private final List<AuthenticationExecutionModel> executions;
     private final AuthenticationProcessor processor;
     private final AuthenticationFlowModel flow;
-    private boolean successful;
+    private boolean successful = false;
```

Comment:
May I suggest Boolean.FALSE?

Reply:
Hm... I am not sure if it is better to use `Boolean.FALSE` ? This would mean one more necessity for auto-boxing from Boolean object (`Boolean.FALSE`) to primitive value `false`. And the `successful` is a primitive variable, which is not sent anywhere and not parsed from/to String etc and hence is completely fine with stick only to primitive value (true/false) IMO without need of any boxing/unboxing from/to Java object of type `Boolean`? IMO best for performance and also readability.

Or do you have some article or resource, which recommends to not use primitive for such use-case?

Reply:
It seems fine to me, just following some guides to avoid constants. But not any problem or technical need.

Reply:
Thanks for clarification. I consider constants makes sense especially for strings and numbers. Also for example when I want to specify for example default value or when I want to search for the usage. Pretty much most of the use-cases like used for example in class `org.keycloak.models.Constants` . For this particular use-case, I don't see much value of using `Boolean.FALSE` instead of `false` . But thanks for bringing this.